File: string/string_python/length_of_string.py
#1. Find the length of a string.
n = input("Enter a string:")

print(len(n))

#2. Check if a string is empty.
if n == "":
    print("String is empty")
else:
    print("String is not empty")

#3. Check if a string has exactly 5 characters.
if len(n) == 5:
    print("The string has exactly 5 characters.") 
else:
    print("The string does not have exactly 5 characters.")

#4. Check if the string length is even or odd.
if len(n) % 2 == 0:
    print("Even")
else:
    print("Odd")

#5. Check if the string has more than 10 characters.
if len(n) >= 10:
    print("String has more than 10 characters")
else:
    print("String dont have more than 10 characters")

#6. Print the last index of a string.
last_index = len(n) -1
print("last index:", last_index)

#7. Print the first half of a string.
half = len(n) // 2
print("First half Character:",n[:half])

#8. Print the second half of a string.
print("First half Character:",n[half:])

#9. Print the middle character of a string.
middle = len(n) // 2
print("middle character",n[middle])

#10. Check whether two strings have the same length.
str1 = "Dog"
str2 = "cat"

if len(str1) == len(str2):
    print("Both string have same length")
else:
    print("Both string do not  have same length")

File: string/string_js/length_of_string.js
//1. Find the length of a string.
let a = "Butterfly"
console.log(a.length);

//2. Check if a string is empty.
let b = ""
if (b == "") {
    console.log("String is empty")
} else {
    console.log("String does not empty")
}

//3. Check if a string has exactly 5 characters.
let c = "mohit"
if (c.length == 5) {
    console.log("The string has exactly 5 character");
} else {
    console.log("The string do not have exactly 5 character");
}

//4. Check if the string length is even or odd.
let d = "Birds";
let len = d.length;
if (len % 2 == 0) {
    console.log( "Even");
} else {
    console.log( "Odd");
}

//5. Check if the string has more than 10 characters.
let e = "Programm"
if (e.length > 10) {
    console.log("The string has more than 10 characters.");
} else {
    console.log("He string has 10 or fewer characters.")
}


//6. Print the last index of a string. 
let f = "JavaScript"
let lastIndex = f.length -1;
console.log("Last Index:", lastIndex);

//7. Print the first half of a string.
let g = "Hello"
let half = g.slice(0, g.length / 2);
console.log("First half:", half);

//8. Print the second half of a string.
let secondHalf = g.slice(g.length / 2);
console.log("Second half:", secondHalf);

//9. Print the middle character of a string.
let middleCharacter = Math.floor(g.length / 2);
console.log("Middle character:", g[middleCharacter]);

//10. Check whether two strings have the same length.
let strOne = "Dog";
let strTwo = "Cat";

if (strOne.length == strTwo.length) {
    console.log("Both string have same length")
} else {
    console.log("Both string don't have same length")
}



